internal/services/recoveryservices/site_recovery_replication_policy_hyperv_resource.go

```diff
@@ -0,0 +1,244 @@
+			}
+
+			state := ReplicationPolicyHyperVModel{
+				Name:              id.ReplicationPolicyName,
+				ResourceGroupName: id.ResourceGroupName,
+				RecoveryVaultName: id.VaultName,
+			}
+
+			if model := resp.Model; model != nil {
+				if detail, isA2A := expandH2APolicyDetail(resp.Model); isA2A {
+					if detail.ApplicationConsistentSnapshotFrequencyInHours != nil {
+						state.ApplicationConsistentSnapshotFrequencyInHours = *detail.ApplicationConsistentSnapshotFrequencyInHours
+					}
+					if detail.ReplicationInterval != nil {
+						state.CopyFrequency = *detail.ReplicationInterval
+					}
+					if detail.RecoveryPointHistoryDurationInHours != nil {
+						state.RecoveryPointRetentionInHours = *detail.RecoveryPointHistoryDurationInHours
+					}
+
+				}
+			}
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r ReplicationPolicyHyperVResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ReplicationPolicyHyperVModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			id, err := replicationpolicies.ParseReplicationPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %s: %+v", metadata.ResourceData.Id(), err)
+			}
+
+			client := metadata.Client.RecoveryServices.ReplicationPoliciesClient
+
+			parameters := replicationpolicies.UpdatePolicyInput{
+				Properties: &replicationpolicies.UpdatePolicyInputProperties{
+					ReplicationProviderSettings: &replicationpolicies.HyperVReplicaAzurePolicyInput{
+						RecoveryPointHistoryDuration:                  &model.RecoveryPointRetentionInHours,
+						ApplicationConsistentSnapshotFrequencyInHours: &model.ApplicationConsistentSnapshotFrequencyInHours,
+						ReplicationInterval:                           &model.CopyFrequency,
+					},
+				},
+			}
+
+			err = client.UpdateThenPoll(ctx, *id, parameters)
+			if err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ReplicationPolicyHyperVResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			id, err := replicationpolicies.ParseReplicationPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := metadata.Client.RecoveryServices.ReplicationPoliciesClient
+
+			err = client.DeleteThenPoll(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("deleting  %s : %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ReplicationPolicyHyperVResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return replicationpolicies.ValidateReplicationPolicyID
+}
+
+func expandH2APolicyDetail(input *replicationpolicies.Policy) (out *replicationpolicies.HyperVReplicaAzurePolicyDetails, isA2A bool) {
```

Comment:
What is H2A and A2A?

Reply:
have renamed this func to `expandHyperVToAzurePolicyDetail`, and H2A means HyperV To Azure, A2A means Azure To Azure